### DISCUSSION TITLE
Cancel option when prompted to save or discard changes

### DISCUSSION THREAD

**Discussion by jtruland** [mealie-recipes/mealie#2899] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

I accidentally clicked Close while editing a recipe.  I wanted to continue working, not save my changes exactly as they were, nor fully discard all of my changes to that point.

### Please provide a concise description of the feature that would resolve your issue.

Add an additional option "Cancel" to Save/Discard prompts to return to the current screen and continue editing.

### Please consider and list out some caveats or tradeoffs made in your design decision

None

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by eneiss** [comment]
Up, I'm having the same frustration when using the bulk-edit mode.
I start typing stuff and at some point I misclick outside the floating window, and all my changes are discarded.
It's happened to me a couple times already, so I always end up writing everything on a notepad instead and then copy-paste the final content to Mealie all at once.
